tests/unit/test_virtual_serialize.cc

```diff
@@ -365,10 +377,15 @@ struct TestWrapper {
         vec.emplace_back(std::make_shared<TestDerived3>(TEST_CONSTRUCT{}));
       }
     }
+
+    for (std::size_t i = 0; i < vec_num_elms; i++) {
+      vec_derived.emplace_back(std::make_shared<TestDerived1>(TEST_CONSTRUCT{}));
```

Comment:
It would perhaps be good to use `TestDerived2` and `TestDerived3` in exercising the case we're concerned about, since 2 is actually intermediate between `TestBase` and 3

Reply:
Done